### PR TITLE
feat(webhooks): Support preconfigured webhook stages

### PR DESCRIPTION
**Pull request by jervi**
Adds support for preconfigured webhook stages that appears as native stages in Deck. Please have a look, I'm sure there's a lot of room for improvement here 😃

Screenshots from Deck:
skjermbilde 2017-05-09 kl 15 08 39
skjermbilde 2017-05-09 kl 15 09 55

Example configuration in `orca.yml`:

```yaml
webhook:
  preconfigured:
    - label: Some Webhook
      description: This is a webhook stage, but it appears as a native stage in Spinnaker
      type: customWebhook # Should be unique
      # The following properties are all optional:
      enabled: true # default true
      url: https://my.webhook.com/event
      headers:
        Accept:
          - application/json
        AnotherHeader:
          - valueOne
          - valueTwo
      method: POST
      payload: |
        {
          "text": "Version ${trigger.buildInfo.artifacts[0].version} deployed"
        }
      waitForCompletion: true
      # The rest of the properties are only used if waitForCompletion == true
      statusUrlResolution: webhookResponse # getMethod, locationHeader, webhookResponse
      statusUrlJsonPath: $.statusUrl # Only used if statusUrlResolution == webhookResponse
      statusJsonPath: $.status
      progressJsonPath: $.progress
      successStatuses: SUCCESS,COMPLETED
      canceledStatuses: CANCELED
      terminalStatuses: TERMINATED
    - label: Another Webhook
      description: This is also a webhook stage, but it has no properties set
      type: anotherCustomWebhook
      enabled: false
```

The properties that are defined in Orca are not surfaced to Deck in the configuration stage, however, all properties are visible in the execution context, so there should be no secrets in the configuration for now. I'm considering a way to mitigate this.

This PR adds a new endpoint that is queried from Deck that just pass the relevant parts of the configuration to the frontend, and a new `PreconfiguredWebhookStage` that extends `WebhookStage`. This new stage just merge the configuration in the stage context with the preconfigured values.

Corresponds with PR https://github.com/spinnaker/gate/pull/389 and https://github.com/spinnaker/deck/pull/3655.
See issue https://github.com/spinnaker/spinnaker/issues/1512.

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
LGTM - any concerns @robfletcher or @cfieber ?

**Comment by robfletcher** [comment]
My only concern is around the changes made to `ExecutionRunnerSupport`. That class will almost certainly be going away once we complete the migration to the v3 execution engine. Equivalent changes will need to be in `StartStageHandler` going forward.

**Comment by jervi** [comment]
@robfletcher I've updated the PR, can you take another look?

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StageBuilderAware.kt [hunk 1]
```diff
@@ -25,6 +25,6 @@ interface StageBuilderAware {
   val stageDefinitionBuilders: Collection<StageDefinitionBuilder>
 
   fun Stage<*>.builder(): StageDefinitionBuilder =
-    stageDefinitionBuilders.find { it.type == getType() }
+    stageDefinitionBuilders.find { it.type == getType() || getContext()?.get("alias") == it.type }
```
Comment (robfletcher) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StageBuilderAware.kt]: Can `getContext()` actually return `null`? I would hope not.

**Comment by robfletcher** [comment]
Looks right to me (apart from minor point above). Appreciate the test.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StageBuilderAware.kt (hunk 1 above)
Comment (jervi) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StageBuilderAware.kt]: I've removed the null-safe operator and rebased the branch

**Comment by jervi** [comment]
@robfletcher @anotherchrisberry Can we get this and the corresponding gate and deck PR merged?

**Comment by Nagarajj** [comment]
@jervi @anotherchrisberry We have some secrets in the WebHook Stage Payload which is displayed in the UI as properties in the execution context. 

Any ideas around around how to restrict it being available in the UI. Thanks for all the help in advance.

**Comment by willgorman** [comment]
We have a similar concern, although with headers instead of the payload.  While those don't show up directly in the UI they are still visible when viewing the source of the execution context for the pipeline.  I'm curious what mitigation you were considering @jervi?

**Pull request merged**